### PR TITLE
Update to colours to change on dark mode.

### PR DESCRIPTION
**Pull request by kelsiesmurphy**
- [Description](#description)
  - [Type of change](#type-of-change)
- [Testing locally](#testing-locally)
- [Checklist](#checklist)

# Description

### Problem
The segmented picker (See the Testing locally section for the locations of this component) did not update to dark mode colours in dark mode, which is different to the Figma design.

### Solution
Added the styling changes necessary to make this happen. In some cases where there is no preset colour style in the NativeBase styling, I used a hex colour value inline in the code. This was either because the element (divider in this case) had opacity styling and this was the simplest solution, or it was a one time colour use. Happy to update if required.

Fixes # SVA-307

## Type of change

- [x] New Feature (non-breaking change which adds functionality)

# Testing locally

- Load up the app in an emulator or similar, and check that the segmented picker updates its colours when changed between dark and light mode, matching the Figma design.
- The three places the segmented picker shows are:
1. On the projects list page, showing the options: 'All/Saved/My'
2. On the Settings page, showing the options: 'System/Dark/Light'
3. When you click the search icon in the upper right, showing the options: 'Roles/Tech/Causes'

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have removed any unnecessary comments or console logging
- [ ] I have made corresponding changes to the documentation (if required)
- [x] I have addressed accessibility, if needed
- [x] I have followed best practices, e.g. NativeBase approaches and theming
- [x] I have checked the app in dark mode, if making front-end design changes
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by M4RC1N76** [comment]
HI @kelsiesmurphy , I am testing your SVA-307 ticket and  the Settings are fine, search icon too (nice work!!). I have came across the issue of the emulator being stuck on the Project List skeleton view. Is it only my local issue or something else. Could you advise on that, please?

**Comment by kelsiesmurphy** [comment]
Hey @M4RC1N76, three questions on that:
1. Could you advise on the steps to reproduce this issue?
2. Does it only happen on the SVA-307 branch or other branches also?
3. If you force quit the app in your emulator and relaunch it, does it continue to happen, or does it work fine then?

**Comment by DougieWougie** [comment]
> HI @kelsiesmurphy , I am testing your SVA-307 ticket and the Settings are fine, search icon too (nice work!!). I have came across the issue of the emulator being stuck on the Project List skeleton view. Is it only my local issue or something else. Could you advise on that, please?

I'm not seeing that on my local copy of this PR @M4RC1N76 but the API has been getting spammed by bots recently.

**Review by DougieWougie — APPROVED** [review]
Checked on Android, thanks @kelsiesmurphy !

File: app/src/NativeBase/Components/SegmentedPicker.tsx [hunk 1]
```diff
@@ -44,15 +45,25 @@ const SegmentedPicker: FC<SegmentedPickerProps> = ({
         <Pressable flexGrow="1" onPress={option.onPress}>
           <Box
             alignItems="center"
-            backgroundColor={option.isSelected ? 'white' : 'transparent'}
+            _dark={{
+              backgroundColor: option.isSelected ? '#636366' : 'transparent',
```
Comment (joefhall) [inline app/src/NativeBase/Components/SegmentedPicker.tsx:49]: Please can you put all of the new colour values into StaTheme.tsx and reference them using colour names -- that's the NativeBase theming approach, plus it's more helpful to see a colour name, anyone can more quickly see what it is.  

As a rule we want to keep all colour definitions in the theme file, and these colours may well need to be used in other components in the future, so defining them there makes that possible.

Please see https://github.com/Scottish-Tech-Army/Volunteer-app/blob/main/APP_DEVELOPMENT.md#theming--approach

**Review by joefhall — CHANGES_REQUESTED** [review]
Thanks @kelsiesmurphy looks good visually.

Be good to move these colours into the theme file, please see comment.

File: app/src/NativeBase/Components/SegmentedPicker.tsx (hunk 1 above)
Comment (kelsiesmurphy) [reply app/src/NativeBase/Components/SegmentedPicker.tsx:49]: Will sort this tomorrow, thanks! Is there a clever way to deal with the transparent hex values in the divider, or should it just be a 8 digit hex value in the theme?

Comment (joefhall) [reply app/src/NativeBase/Components/SegmentedPicker.tsx:49]: Great, thanks. Nothing different to do for semi transparent colours that I know of. I’m assuming there definitely needs to be transparency set on those? (they can’t just be normal solid colours?)

Comment (joefhall) [reply app/src/NativeBase/Components/SegmentedPicker.tsx:49]: (A solid background colour would probably give slightly better performance)

Comment (kelsiesmurphy) [reply app/src/NativeBase/Components/SegmentedPicker.tsx]: [This is now done I think](https://github.com/Scottish-Tech-Army/Volunteer-app/pull/158/commits/bc10609558b14e28d4f72de0b38956a38a69be36). I made them solid colours.

**Review by joefhall — APPROVED** [review]
Nice one, thanks @kelsiesmurphy -- looks good!

**Pull request merged**